lms/djangoapps/ccx/views.py

```diff
@@ -258,6 +258,17 @@ def override_fields(parent, data, graded, earliest=None, ccx_ids_to_delete=None)
                 graded[block.format] = graded.get(block.format, 0) + 1
 
             children = unit.get('children', None)
+            # For a vertical override start and due dates of its all problems.
+            if unit['category'] == u'vertical':
```

Comment:
`if unit.get('category', None) == u'vertical'`

this is safer (you provide a default value JUST IN CASE there's not a category).